provider/kubernetes/kubernetes.go

```diff
@@ -636,37 +648,54 @@ func getRuleForHost(host string) string {
 	return "Host:" + host
 }
 
-func getTLS(ingress *extensionsv1beta1.Ingress, k8sClient Client) ([]*tls.Configuration, error) {
-	var tlsConfigs []*tls.Configuration
+func getTLS(ingress *extensionsv1beta1.Ingress, k8sClient Client, tlsConfigs map[string]*tls.Configuration) error {
 
 	for _, t := range ingress.Spec.TLS {
 		secret, exists, err := k8sClient.GetSecret(ingress.Namespace, t.SecretName)
 		if err != nil {
-			return nil, fmt.Errorf("failed to fetch secret %s/%s: %v", ingress.Namespace, t.SecretName, err)
+			return fmt.Errorf("failed to fetch secret %s/%s: %v", ingress.Namespace, t.SecretName, err)
 		}
 		if !exists {
-			return nil, fmt.Errorf("secret %s/%s does not exist", ingress.Namespace, t.SecretName)
+			return fmt.Errorf("secret %s/%s does not exist", ingress.Namespace, t.SecretName)
 		}
 
 		cert, key, err := getCertificateBlocks(secret, ingress.Namespace, t.SecretName)
 		if err != nil {
-			return nil, err
+			return err
 		}
 
-		entryPoints := getSliceStringValue(ingress.Annotations, annotationKubernetesFrontendEntryPoints)
+		newEntryPoints := getSliceStringValue(ingress.Annotations, annotationKubernetesFrontendEntryPoints)
 
-		tlsConfig := &tls.Configuration{
-			EntryPoints: entryPoints,
-			Certificate: &tls.Certificate{
-				CertFile: tls.FileOrContent(cert),
-				KeyFile:  tls.FileOrContent(key),
-			},
+		tlsConfig, tlsExists := tlsConfigs[t.SecretName]
```

Comment:
Using Secret as a map key is wrong you need to use Namespace + SecretName as you might have same secret names in different namespaces.